### ISSUE TITLE
error during compilation

### ISSUE THREAD

**Issue by fuzun** [pureqml/qmlcore#191]
I have come up to this project and found it interesting.

I tried to follow the "Simple how-to" procedure to see how it works but I get this error:

```
building app for web...
...
parsing ./qmlcore\core\ScrollView.qml ... core.ScrollView
parsing ./qmlcore\core\SequentialAnimation.qml ... core.SequentialAnimation
parsing ./qmlcore\core\Shadow.qml ... core.Shadow
parsing ./qmlcore\core\System.qml ... core.System
parsing ./qmlcore\core\Text.qml ... core.Text
parsing ./qmlcore\core\Timer.qml ... core.Timer
parsing ./qmlcore\core\Transform.qml ... core.Transform
parsing ./qmlcore\core\VideoPlayer.qml ... core.VideoPlayer
parsing ./qmlcore\core\WheelMixin.qml ... core.WheelMixin
**error: 'charmap' codec can't decode byte 0x9d in position 23994: character maps to <undefined>**
```

Note that I have not modified any files.

**Comment by whoozle** [comment]
it looks like python can't use your locale for some reason. can you show me `build -v` output please? Did you change main qml file at all?

**Comment by fuzun** [comment]
Log:
```
building app for web...
running using 1 jobs
including platform initialisation file at ./qmlcore\platform\html5\.core.js
including platform initialisation file at ./qmlcore\platform\video.html5\.core.js
including platform initialisation file at ./qmlcore\platform\web\.core.js
including js file... ./qmlcore\core\core.js
including js file... ./qmlcore\core\gradient.js
including js file... ./qmlcore\core\model.js
including js file... ./qmlcore\core\transform.js
including js file... ./qmlcore\platform\html5\cache.js
error: 'charmap' codec can't decode byte 0x9d in position 23994: character maps to <undefined>
error: 'charmap' codec can't decode byte 0x9d in position 23994: character maps to <undefined>
Traceback (most recent call last):
  File "./qmlcore/build", line 436, in <module>
    build(root, platform, apps, app, manifest)
  File "./qmlcore/build", line 330, in build
    compile_qml(target, root, project_dirs, manifest, app, **kw)
  File "C:\Projects\pureqml-test\qmlcore\compiler\__init__.py", line 312, in compile_qml
    c.generate()
  File "C:\Projects\pureqml-test\qmlcore\compiler\__init__.py", line 201, in generate
    self.process_files(None, generator)
  File "C:\Projects\pureqml-test\qmlcore\compiler\__init__.py", line 158, in process_files
    promise = self.process_file(pool, generator, package, dirpath, filename)
  File "C:\Projects\pureqml-test\qmlcore\compiler\__init__.py", line 94, in process_file
    data = f.read()
  File "C:\Python38\lib\encodings\cp1252.py", line 23, in decode
    return codecs.charmap_decode(input,self.errors,decoding_table)[0]
UnicodeDecodeError: 'charmap' codec can't decode byte 0x9d in position 23994: character maps to <undefined>
```

By the way I'm on Windows.

I did not change any file. I just cloned the repository and tried to follow the simple how to.

**Comment by fuzun** [comment]
@whoozie,

Adding ` encoding='utf-8' ` parameter to open() functions in __init.py__ at line 231 and 93 seems to fixed the problem. You might want to create a patch for this.

**Comment by marechenok** [comment]
@fuzun thank you. solved my problem

**Comment by whoozle** [comment]
fixed in 875357b1ff20b5e4827fedde506d72883519eb0e
please check
